web/src/lib/components/shared-components/context-menu/context-menu.svelte

```diff
@@ -22,7 +22,7 @@
 		if (!menuEl) return;
 
 		const rect = menuEl.getBoundingClientRect();
-		x = Math.min(window.innerWidth - rect.width, x);
+		x = Math.min(window.innerWidth - rect.width, x - rect.width);
```

Comment:
The best and most elegant solution in my opinion would be confining it to the current container. This way it would be easy to automatically determine where to display it. In this case, it would still not work because the container includes the information panel and without a major rework that is not easily modifiable.

Another solution I can think of would be to specialize this instance of the context menu with a prop that defaults to a value
Yet another option would be to display it over everything else(which seems to have been the intention from the start given its z-index), however z-index is context dependent when the parent is using absolute positioning